### ISSUE TITLE
Unable to reproduce

### ISSUE THREAD

**Issue by overnap** [kaistmm/seed-pytorch#5]
Hi.

I'm having trouble training the model with the current commit following the instructions in the `README.md`. There are quite a few errors, mostly typos and path issues.

For example, if I set up the data as described in `datasets/README.md`, the paths don't match what `main.py` actually looks for. I managed to fix these and a few other critical bugs on my own to get it running, but I've hit a wall with the results.

Even with the provided config, I cannot reproduce the performance metrics reported in the paper. I can get close results only when using the pretrained weights, but training diffusion from scratch doesn't seem to work as expected.

What are the exact hyperparameter settings used to achieve the results in the paper? Is there something missing in the current config `ECAPA_SEED_rdmmlp3.yaml`?

Thanks.

**Comment by DevKiHyun** [comment]
Hi @overnap ,

Thank you for your interest in our work and for taking the time to report these issues. 
We apologize for the typos and path inconsistencies you encountered while setting up the repository.

Regarding the dataset preparation, the setup can sometimes vary depending on your local environment. The datasets/README.md is intended to be an informal guide to help users access the datasets we used for our experiments. As long as you can prepare the datasets yourself and configure the recommended paths and file lists, the training script should run, even if it requires a few minor manual adjustments on your end.

To help us figure out why the performance isn't matching the paper, could you provide a bit more detail on how the results diverged? Sharing your training logs and the exact metrics you are seeing would be incredibly helpful.

Please also keep in mind that because SEED relies on a diffusion model, there is an inherent level of uncertainty and variance when training from scratch. You may need to try a few different runs, as we actually experienced similar fluctuations during our own initial experiments.

Would you be able to upload your training logs here so we can take a closer look and help you troubleshoot?

**Comment by overnap** [comment]
As I mentioned, I haven't modified the training code or the configurations, except for a few essential changes required for execution, such as data path structures.

The logs start as follows:

```
Ignored unknown parameter trainfunc in yaml.
Ignored unknown parameter use_speaker_label in yaml.
Python Version:  3.11.13 (main, Jun 29 2025, 16:24:30) [Clang 20.1.4 ]
PyTorch Version: 2.1.2+cu121
Number of GPUs:  4
Save path:       exps/ecapa_seed_rdmmlp3
Initialised AdamW optimizer
Initialised step LR scheduler
Diffusion use: True
Diffusion training  timesteps: 1000
Diffusion inference timesteps: 50
Diffusion with condition: False
Diffusion self condition: True
Diffusion training clamp norm: False
Diffusion init x0  scale:      False
Diffusion sampling clamp norm: False
Diffusion sampling final norm: False
Diffusion normalize type:      l2
Diffusion sampling by DDIM:    False
Diffusion feature ensemble:    False
Diffusion network ema value:   0
Activate Audio 8k augmentation: False
Max Learning rate : 0.0005
Seed value: 33
Total parameters:      17.29M
Trainable parameters:  2.44M (Diffusion network)
Frozen parameters:     14.86M
ECAPA_TDNN Model loaded from ./pretrained/official_ecapatdnn.model
Unexpected keys (4) from official_ecapatdnn.model:
  - __L__.softmax.fc.weight
  - __L__.softmax.fc.bias
  - __L__.angleproto.w
  - __L__.angleproto.b
```

At around epoch 60 (as reported in the paper), the scores I'm getting are:

```
Epoch 58, VEER 0.9623, MinDCF 0.06822 TEST: vox1-O
Epoch 58, VEER 3.1599, MinDCF 0.26509 TEST: vcmix_test
Epoch 59, TEER/TAcc 0.00, TLOSS 1.650528, LR 0.000500 
Epoch 59, VEER 0.9570, MinDCF 0.06748 TEST: vox1-O
Epoch 59, VEER 2.8941, MinDCF 0.25324 TEST: vcmix_test
Epoch 60, TEER/TAcc 0.00, TLOSS 1.651084, LR 0.000500 
Epoch 60, VEER 0.9839, MinDCF 0.06956 TEST: vox1-O
Epoch 60, VEER 3.0334, MinDCF 0.25961 TEST: vcmix_test
Epoch 61, TEER/TAcc 0.00, TLOSS 1.651106, LR 0.000500 
Epoch 61, VEER 0.9467, MinDCF 0.06679 TEST: vox1-O
Epoch 61, VEER 2.8774, MinDCF 0.24409 TEST: vcmix_test
Epoch 62, TEER/TAcc 0.00, TLOSS 1.649129, LR 0.000500 
Epoch 62, VEER 0.9467, MinDCF 0.07009 TEST: vox1-O
Epoch 62, VEER 3.0795, MinDCF 0.25967 TEST: vcmix_test
Epoch 63, TEER/TAcc 0.00, TLOSS 1.650376, LR 0.000500 
Epoch 63, VEER 0.9145, MinDCF 0.07349 TEST: vox1-O
Epoch 63, VEER 2.9678, MinDCF 0.26248 TEST: vcmix_test
Epoch 64, TEER/TAcc 0.00, TLOSS 1.651010, LR 0.000500 
Epoch 64, VEER 0.9573, MinDCF 0.06828 TEST: vox1-O
Epoch 64, VEER 2.9375, MinDCF 0.25503 TEST: vcmix_test
Epoch 65, TEER/TAcc 0.00, TLOSS 1.650562, LR 0.000500 
Epoch 65, VEER 0.9361, MinDCF 0.07136 TEST: vox1-O
Epoch 65, VEER 3.0121, MinDCF 0.25971 TEST: vcmix_test
Epoch 66, TEER/TAcc 0.00, TLOSS 1.650153, LR 0.000500 
Epoch 66, VEER 0.9627, MinDCF 0.06642 TEST: vox1-O
Epoch 66, VEER 2.8561, MinDCF 0.24876 TEST: vcmix_test
Epoch 67, TEER/TAcc 0.00, TLOSS 1.649648, LR 0.000500 
Epoch 67, VEER 0.9414, MinDCF 0.07131 TEST: vox1-O
Epoch 67, VEER 2.9467, MinDCF 0.25352 TEST: vcmix_test
Epoch 68, TEER/TAcc 0.00, TLOSS 1.649403, LR 0.000500 
Epoch 68, VEER 0.9357, MinDCF 0.06982 TEST: vox1-O
Epoch 68, VEER 2.8561, MinDCF 0.25029 TEST: vcmix_test
Epoch 69, TEER/TAcc 0.00, TLOSS 1.649034, LR 0.000500 
Epoch 69, VEER 0.9095, MinDCF 0.06945 TEST: vox1-O
Epoch 69, VEER 3.0320, MinDCF 0.25925 TEST: vcmix_test
Epoch 70, TEER/TAcc 0.00, TLOSS 1.650211, LR 0.000500 
Epoch 70, VEER 0.9999, MinDCF 0.07211 TEST: vox1-O
Epoch 70, VEER 3.0156, MinDCF 0.25804 TEST: vcmix_test
```

Given the nature of diffusion models, I'm aware that performance can vary based on the seed. The provided scores are from the training phase; using different seeds in eval mode did yield better results (EER ~0.88 on Vox1), but they still fall short of the impressive benchmarks stated in the paper, especially on VoxSRC23 or VC-Mix.
